File: Homeworks/C#2/StringsAndTextProcessingHomework/02.ReverseString/ReverseString.cs
/*
 Problem 2. Reverse string

Write a program that reads a string, reverses it and prints the result at the console.
Example:

input	output
sample	elpmas
*/ 


using System;

namespace ReverseStringProblem
{
    class ReverseString
    {
        public static string ReverseWhitLoop(string text)
        {
            char[] charArray = text.ToCharArray();
            string reverse = String.Empty;
            for (int i = charArray.Length - 1; i > -1; i--)
            {
                reverse += charArray[i];
            }
            return reverse;
        }

        public static string Reverse(string text)
        {
            char[] charArray = text.ToCharArray();
            Array.Reverse(charArray);
            return new string(charArray);
        }

        static void Main()
        {
            Console.Write("Please, enter some string: ");
            string input = Console.ReadLine();

            Console.WriteLine("Reversed string with loop is: " + ReverseWhitLoop(input));

            Console.WriteLine("Reversed string is: " + Reverse(input));
        }
    }
}

File: Homeworks/JavaScriptFundamentals/08.Strings/01.ReverseString/ReverseString.js
/*
 Write a JavaScript function that reverses a string and returns it.
 Example:

 input	output
 sample	elpmas
 */

var jsConsole,
    button = document.getElementById('reverse');

button.onclick = reverseString;

function reverseString() {
    var input = jsConsole.read('#input');
    var reveredInput =  [].map.call(input, function (x) {return x;}).reverse().join('');

    jsConsole.writeLine('Input string is: ' + input);
    jsConsole.writeLine('Reversed input string is: ' + reveredInput);

    console.log('Input string is: ' + input);
    console.log('Reversed input string is: ' + reveredInput);
}

